vagrant/ansible/roles/samba.setup/tasks/main.yml

```diff
@@ -36,16 +36,18 @@
         name: 'sit.{{ config.be.name }}'
         tasks_from: new_share.yml
       vars:
-        name: "{{ item.share_name }}"
+        name: "{{ item.share_name }}-{{ config.be.name }}-{{ config.be.variant }}"
         path: "{{ item.cluster_volume }}"
-        file: "/etc/samba/smb.shares/{{ item.share_name }}.conf"
+        file: "/etc/samba/smb.shares/{{ name }}.conf"
       with_items: "{{ samba_shares }}"
 
     - name: Include share configuration into smb.conf
       lineinfile:
         path: "/etc/samba/smb.conf"
-        line: "include = /etc/samba/smb.shares/{{ item.share_name }}.conf"
+        line: "include = /etc/samba/smb.shares/{{ name }}.conf"
         insertafter: EOF
+      vars:
+        name: "{{ item.share_name }}-{{ config.be.name }}-{{ config.be.variant }}"
```

Comment:
Maybe it could be cleaner to set this var at the _block_ level and use it in all subtasks.

Reply:
Done.